### PR TITLE
Check share status when touching versions

### PR DESCRIPTION
**Pull request by artonge**

### CODE REVIEW

**Review by susnux — APPROVED** [review]

File: apps/files_versions/src/components/Version.vue [hunk 1]
```diff
@@ -106,6 +107,9 @@ import { translate as t } from '@nextcloud/l10n'
 import { joinPaths } from '@nextcloud/paths'
 import { getRootUrl } from '@nextcloud/router'
 import { loadState } from '@nextcloud/initial-state'
+import { Permission } from '@nextcloud/files'
+
+import { hasPermissions } from '../../../files_sharing/src/lib/SharePermissionsToolBox.js'
```
Comment (skjnldsv) [inline apps/files_versions/src/components/Version.vue:112]: :see_no_evil:

Comment (skjnldsv) [reply apps/files_versions/src/components/Version.vue:112]: Not really better than `(this.fileInfo.permissions & Permission.UPDATE) !== 0`

**Review by skjnldsv — APPROVED** [review]

File: apps/files_versions/src/components/Version.vue (hunk 1 above)
Comment (skjnldsv) [reply apps/files_versions/src/components/Version.vue:112]: Right, I should have blocked it

Comment (skjnldsv) [reply apps/files_versions/src/components/Version.vue:112]: https://github.com/nextcloud/server/pull/43754

**Comment by artonge** [comment]
/backport 8e95d0f3ae4fb9e186e3995cd2aa7d8393d7003d to stable25

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc, 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable25

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc, 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable24

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc, 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable23

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable25

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc, 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable24

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc, 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable233

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc, 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable23

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable24

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable25

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable24

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable23

**Comment by artonge** [comment]
/backport 741dec283fe1aa3e19a9b525a1a8abc70261a5dc 37d0fd109a921d760d621a4f0a11cf05c9c7211f to stable23

**Pull request merged**